src/v/config/tests/config_store_test.cc

```diff
@@ -530,3 +532,22 @@ SEASTAR_THREAD_TEST_CASE(property_aliasing) {
 
     BOOST_TEST(cfg.property_names_and_aliases().contains("aliased_bool"));
 }
+
+SEASTAR_THREAD_TEST_CASE(ignored_keys) {
+    auto yaml_with_prop_to_ignore = YAML::Load(R"yaml(
+secret_string: terces
+aliased_bool_legacy: false
+    )yaml");
+
+    // smoke test: check that the properties are valid for test_config
+    auto cfg = test_config{};
+
+    BOOST_REQUIRE_NO_THROW(cfg.read_yaml(yaml_with_prop_to_ignore));
+    BOOST_CHECK(cfg.secret_string.value() == "terces");
+    BOOST_CHECK(cfg.aliased_bool.value() == false);
+
+    // actual test: check that noop_cfg can ignore properties for test_config
+    auto noop_cfg = noop_config{};
+    BOOST_REQUIRE_NO_THROW(noop_cfg.read_yaml(
```

Comment:
ah but that's not how it works: the parameter is called ignored_missing: if the yaml key is not found in the properties of the caller, then it will throw std::illegal_argument unless it's in ignore_missing.

I'll add an extension to show this

Reply:
Ah, much clearer now! Thank you!